From 8661e9dc65c3361af9e5ed7a5f5d378de27b2180 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Fri, 13 Feb 2015 11:58:19 +0530
Subject: [PATCH 011/132] edac/octeon_edac-lmc: Fix kernel panic when 1 DDR
 present

Commit 6c66d68ff92acb6f1a2251d85d0d6ddd3b1ba871 from
git://git.yoctoproject.org/linux-yocto-3.14

Add check to see if DDR is available.

Signed-off-by: David Daney <david.daney@cavium.com>
Signed-off-by: Prem Mallappa <pmallappa@caviumnetworks.com>
Signed-off-by: Abhishek Paliwal <abhishek.paliwal@aricent.com>
[Original patch taken from OCTEON-SDK 3.1.1-544.]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/edac/octeon_edac-lmc.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/edac/octeon_edac-lmc.c b/drivers/edac/octeon_edac-lmc.c
index c9b2f9e..782eac5 100644
--- a/drivers/edac/octeon_edac-lmc.c
+++ b/drivers/edac/octeon_edac-lmc.c
@@ -246,6 +246,18 @@ static int octeon_lmc_edac_probe(struct platform_device *pdev)
 
 	if (OCTEON_IS_OCTEON1PLUS()) {
 		union cvmx_lmcx_mem_cfg0 cfg0;
+		cvmx_l2c_cfg_t l2c_cfg;
+		int present = 0;
+
+		l2c_cfg.u64 = cvmx_read_csr(CVMX_L2C_CFG);
+
+		if (mc == 0)
+			present = l2c_cfg.s.dpres0;
+		else
+			present = l2c_cfg.s.dpres1;
+
+		if (!present)
+			return -ENXIO;
 
 		cfg0.u64 = cvmx_read_csr(CVMX_LMCX_MEM_CFG0(0));
 		if (!cfg0.s.ecc_ena) {
-- 
1.9.1

